lib/ansible/modules/remote_management/cpm/cpm_serial_port_info.py

```diff
@@ -0,0 +1,179 @@
+    required: false
+    type: bool
+    default: true
+  use_proxy:
+    description: Flag to control if the lookup will observe HTTP proxy environment variables when present.
+    required: false
+    type: bool
+    default: false
+  port:
+    description:
+      - This is the serial port number that is getting retrieved. It can include a single port
+      - number, multiple port numbers separated by commas or an '*' character for all ports.
+    required: true
+"""
+
+EXAMPLES = """
+# Get Parameters for one Serial Port
+- name: Get the Serial Port Parameters for port 2 of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "2"
+
+# Get Parameters for multiple Serial Ports
+- name: Get the Serial Port Parameters for ports 2 and 4 of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "2,4"
+
+# Get Parameters for All Serial Ports
+- name: Get the Serial Port Parameters for all ports of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "*"
+"""
+
+RETURN = """
+data:
+    description: The output JSON returned from the commands sent
+    returned: always
+    type: str
+"""
+
+import base64
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_bytes, to_native
+from ansible.module_utils.six.moves.urllib.error import HTTPError, URLError
+from ansible.module_utils.urls import open_url, ConnectionError, SSLValidationError
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        cpm_url=dict(type='str', required=True),
+        cpm_username=dict(type='str', required=True),
+        cpm_password=dict(type='str', required=True, no_log=True),
+        port=dict(type='str', required=True),
+        use_https=dict(type='bool', default=True),
+        validate_certs=dict(type='bool', default=True),
+        use_proxy=dict(type='bool', default=False)
+    )
+
+    result = dict(
+        changed=False,
+        data=''
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    auth = to_text(base64.b64encode(to_bytes('{0}:{1}'.format(to_native(module.params['cpm_username']), to_native(module.params['cpm_password'])),
+                   errors='surrogate_or_strict')))
+
+    if module.params['use_https'] is True:
+        protocol = "https://"
+    else:
+        protocol = "http://"
+
+    fullurl = ("%s%s/api/v2/config/serialports?ports=%s" % (protocol, to_native(module.params['cpm_url']), to_native(module.params['port'])))
```

Comment:
If `port` is changed to a `list`, you'll need to do this here:
```suggestion
    ports = module.params['port']
    if isinstance(ports, list):
        ports = ','.join(to_native(x) for x in ports)
    fullurl = ("%s%s/api/v2/config/serialports?ports=%s" % (protocol, to_native(module.params['cpm_url']), ports))
```